ts/src/index.ts

```diff
@@ -118,9 +118,19 @@ export class AutoPush {
     const pushPromises = pushList.map((asset): Promise<void> => {
       return new Promise((resolve, reject) => {
         const pushFile = (pushStream: http2.ServerHttp2Stream): void => {
-          pushStream.on('finish', () => {
+          const onFinish = () => {
+            pushStream.removeListener('error', onError);
             resolve();
-          });
+          };
+          const onError = (err: Error) => {
+            console.error('push stream error for:', asset, err);
```

Comment:
Spurious `console.error`?

Reply:
I want to leave some logs when an error occurs during push. Do you think it should use some logging framework?

Reply:
this library should either emit an event `'pushError'` or accept an option that will be called when this (and maybe other events) happen. fastify has its own logging thing included, see [`req.log`](https://www.fastify.io/docs/latest/Logging/)

Reply:
`'pushError'` sounds good. I'll add that in a separate PR. Will also change the logging with fastify's logging thing. Thanks.

Reply:
Oh, sorry I was confused. This is a common package for auto-push, not specific to fastify. So it can't use fastify's logging.

I'll just add `pushError`.

Reply:
Yeah, this should emit `pushError`, and you should listen to the event in the fastify plugin and log it. You might want to emit the error on the parent stream maybe?

Reply:
`pushStream` is hidden from the client code so it doesn't make sense to emit `pushError` on `pushStream`. I'll emit it on the parent `stream`.